cmd/datasetArchiver/main_test.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"testing"
+)
+
+func TestMainFlagDefaults(t *testing.T) {
+	flag.CommandLine = flag.NewFlagSet("flag defaults", flag.ExitOnError)
+
+	testFlags = func(
+		userpass string,
+		token string,
+		testenvFlag bool,
+		devenvFlag bool,
+		nonInteractiveFlag bool,
+		showVersion bool) {
+		passing := true
+		if userpass != "" {
+			t.Logf("userpass's default should be \"\" but it's \"%s\"", userpass)
+			passing = false
+		}
+		if token != "" {
+			t.Logf("token's default should be \"\" but it's \"%s\"", token)
+			passing = false
+		}
+		if testenvFlag {
+			t.Logf("testenvFlag's default should be false but it's true")
+			passing = false
+		}
+		if devenvFlag {
+			t.Logf("devenvFlag's default should be false but it's true")
+			passing = false
+		}
+		if nonInteractiveFlag {
+			t.Logf("nonInteractiveFlag's default should be false but it's true")
+			passing = false
+		}
+		if showVersion {
+			t.Logf("showVersion's default should be false but it's true")
+			passing = false
+		}
+		if !passing {
+			t.Fail()
+		}
+	}
+
+	os.Args = []string{"test"}
+	main()
+}
+
+func TestMainFlagSetting(t *testing.T) {
```

Comment:
minor comment, up to you if you want to change. One could use loops on tests [here](https://github.com/paulscherrerinstitute/scicat-cli/blob/main/datasetUtils/authenticate_test.go#L33-L79) rather than having multiple functions. If we do this, we could even test more combinations/flags

Reply:
sure that sounds easily feasible.